trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/Bookmark.java

```diff
@@ -0,0 +1,136 @@
+package org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.services;
+
+import java.io.Serializable;
+import java.util.UUID;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Bookmark model for TSP
+ *
+ * @author Kaveh Shahedi
+ * @since 10.1
+ */
+public class Bookmark implements Serializable {
+    private static final long serialVersionUID = -3626414315455912960L;
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private final UUID fUUID;
+    private final String fName;
+    private final String fExperimentId;
+    private final long fStart;
+    private final long fEnd;
+    private final JsonNode fPayload;
+
```

Comment:
In Trace Compass a bookmark could have a color (RGBA). I wonder if we should add that right away. We could add `OutputElementStyle` object here. The FE will then translate the color to `OutputElementStyle`. Or we just keep a single color value.